src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,180 @@
+package duke;
+
+import duke.data.Storage;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class Duke {
+    private static final String LINE = "____________________________________________________________";
+    private static final String ADD_TASK_MSG = "Got it. I've added this duke.task: ";
+    private static final String ERROR_MSG = "☹ OOPS!!! I'm sorry, but I don't know what that means :-(";
+    private static final String TODO_ERROR = "The description of a todo cannot be empty.";
+    private static final String DEADLINE_ERROR = "The description of a deadline cannot be empty and must have a '/by'.";
+    private static final String EVENT_ERROR = "The description of an event cannot be empty and must have a '/at'.";
+    private static final String[] taskTypes = {"todo", "deadline", "event"};
+    private static final String FILE_PATH = "tasks.txt";
+    private static ArrayList<Task> tasks;
+    private static Storage storage;
+
+    public static void main(String[] args) {
+        showHelloGreeting();
+        fileManager();
+        executeResponses();
+        showByeGreeting();
+    }
+
+    private static void fileManager() {
+        try {
+            storage = new Storage(FILE_PATH);
+            tasks = storage.loadTasksFromFile();
+        } catch (DukeException err) {
+            System.out.println("Unable to load file.");
+        }
+    }
+
+    private static void executeResponses() {
+        Scanner in = new Scanner(System.in);
+        String text;
+        text = in.nextLine();
+        while (!text.equals("bye")) {
+            System.out.println(LINE);
+            String[] words = text.split(" ");
+            int numOfTasks = tasks.size();
+            try {
+                switch (words[0]) {
+                case "list":
+                    System.out.println("Here are the tasks in your list:");
+                    for (int i = 0; i < numOfTasks; i++) {
+                        System.out.println((i + 1) + "." + tasks.get(i).toString());
+                    }
+                    break;
+                case "done":
+                    int taskNum = Integer.parseInt(words[words.length - 1]);
```

Comment:
Suggest to relocate expression in parseInt to a seperate statement because it represents a different entity of logic.